src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/LockFileUtils.cs

```diff
@@ -332,9 +332,9 @@ private static void ApplyAliases(string aliases, LockFileItem item)
         private static void AddContentFiles(RestoreTargetGraph targetGraph, LockFileTargetLibrary lockFileLib, NuGetFramework framework, ContentItemCollection contentItems, NuspecReader nuspec, MaccatalystFallback maccatalystFallback)
         {
             // content v2 items
-            var contentFileGroups = contentItems.FindItemGroups(targetGraph.Conventions.Patterns.ContentFiles).ToList();
+            IEnumerable<ContentItemGroup> contentFileGroups = contentItems.FindItemGroups(targetGraph.Conventions.Patterns.ContentFiles);
```

Comment:
This would cause a performance regression for any scenario where more than 1 group is returned, because the methods below would enumerate `contentFileGroups` multiple times.